SECP-service/src/main/resources/assets/app/views/portal/configure.html

```diff
@@ -0,0 +1,113 @@
+    </tr>
+    <tr>
+        <td>Secret</td>
+        <td style="text-align: right"><button ng-click="deletePermission()"><i class="fa fa-trash red" aria-hidden="true"></i></button></td>
+    </tr>
+    </tbody>
+    </table>
+</div>
+
+<div id="rolesTableDiv" style="display: none;">
+    <table id="rolesTable" class="display" cellspacing="0" width="100%">
+        <button class="tableaddbutton" data-toggle="modal" data-target="#roleModal">New Role <i class="fa fa-plus" aria-hidden="true"></i></button>
+        <thead>
+        <tr>
+            <th>Role</th>
+            <th style="text-align: right">Action(s)</th>
+        </tr>
+        </thead>
+        <tfoot>
+        <tr>
+            <th>Role</th>
+            <th style="text-align: right">Action(s)</th>
+        </tr>
+        </tfoot>
+        <tbody>
+        <tr>
+            <td>Developer</td>
+            <td style="text-align: right"><button ng-click="deleteRole()"><i class="fa fa-trash red" aria-hidden="true"></i></button></td>
+        </tr>
+        <tr>
+            <td>Secret</td>
+            <td style="text-align: right"><button ng-click="deleteRole()"><i class="fa fa-trash red" aria-hidden="true"></i></button></td>
+        </tr>
+        </tbody>
+    </table>
+</div>
+
+<!-- Permission Level Modal -->
+<div class="modal fade" id="permissionModal" role="dialog">
+    <div class="modal-dialog">
+
+        <!-- Modal content-->
+        <form name="newpermissionform" class="modal-content" autocomplete="off">
+            <div class="modal-header">
+                <button type="button" class="close" data-dismiss="modal">&times;</button>
+                <h4 class="modal-title">New Permission Level</h4>
+            </div>
+            <div class="modal-body">
+                Permssion Level:<br>
+                <input class="newfilterword" type="text" name="newpermission" required="required" ng-pattern="/^[a-zA-Z0-9]{2,}$/">
+            </div>
+            <div class="modal-footer">
+                <input type="submit" class="btn btn-default" value="Save">
+            </div>
+        </form>
+
+    </div>
+</div>
+
+<!-- Role Modal -->
+<div class="modal fade" id="roleModal" role="dialog">
+    <div class="modal-dialog">
+
+        <!-- Modal content-->
+        <form name="newroleform" class="modal-content" autocomplete="off">
+            <div class="modal-header">
+                <button type="button" class="close" data-dismiss="modal">&times;</button>
+                <h4 class="modal-title">New Role</h4>
+            </div>
+            <div class="modal-body">
+                Role:<br>
+                <input class="newfilterword" type="text" name="newrole" required="required" ng-pattern="/^[a-zA-Z0-9]{2,}$/">
+            </div>
+            <div class="modal-footer">
+                <input type="submit" class="btn btn-default" value="Save">
+            </div>
+        </form>
+
+    </div>
+</div>
+
```

Comment:
You can create these modals dynamically using js. So have a method called levelModal creation, pass in a string (permission or role), in the function go through the steps to create and just pass in the string whenever needed. Then just call that function twice on page load.